robot-command/src/main/java/org/obolibrary/robot/ExpandCommand.java

```diff
@@ -0,0 +1,128 @@
+
+  /** Initialize the command. */
+  public ExpandCommand() {
+    Options o = CommandLineHelper.getCommonOptions();
+    o.addOption(
+        "c",
+        "create-new-ontology",
+        true,
+        "if true, output ontology will only contain the expansions");
+    o.addOption(
+        "a",
+        "annotate-expansion-axioms",
+        true,
+        "if true, annotate expansion axioms with 'prov:wasDerivedFrom <expansion property>'");
+    o.addOption("t", "expand-term", true, "property to expand");
+    o.addOption("T", "expand-term-file", true, "load properties to expand from a file");
+    o.addOption("n", "no-expand-term", true, "properties to not expand");
+    o.addOption("N", "no-expand-term-file", true, "load properties to not expand from a file");
+    o.addOption("i", "input", true, "expand ontology from a file");
+    o.addOption("I", "input-iri", true, "expand ontology from an IRI");
+    o.addOption("o", "output", true, "save expanded ontology to a file");
+    options = o;
+  }
+
+  /**
+   * Name of the command.
+   *
+   * @return name
+   */
+  public String getName() {
+    return "expand";
+  }
+
+  /**
+   * Brief description of the command.
+   *
+   * @return description
+   */
+  public String getDescription() {
+    return "expand ontology";
+  }
+
+  /**
+   * Command-line usage for the command.
+   *
+   * @return usage
+   */
+  public String getUsage() {
+    return "robot expand --input <file> " + "[options] " + "--output <file>";
+  }
+
+  /**
+   * Command-line options for the command.
+   *
+   * @return options
+   */
+  public Options getOptions() {
+    return options;
+  }
+
+  /**
+   * Handle the command-line and file operations for the ExpandOperation.
+   *
+   * @param args strings to use as arguments
+   */
+  public void main(String[] args) {
+    try {
+      execute(null, args);
+    } catch (Exception e) {
+      CommandLineHelper.handleException(e);
+    }
+  }
+
+  /**
+   * Given an input state and command line arguments, expand macro relations, and add axioms to the
+   * input ontology, returning a state with the updated ontology.
+   *
+   * @param state the state from the previous command, or null
+   * @param args the command-line arguments
+   * @return the state with inferred axioms added to the ontology
+   * @throws Exception on any problem
+   */
+  public CommandState execute(CommandState state, String[] args) throws Exception {
+    CommandLine line = CommandLineHelper.getCommandLine(getUsage(), getOptions(), args);
+    if (line == null) {
+      return null;
+    }
+    if (state == null) {
+      state = new CommandState();
+    }
+    IOHelper ioHelper = CommandLineHelper.getIOHelper(line);
+    state = CommandLineHelper.updateInputOntology(ioHelper, state, line);
+    OWLOntology ontology = state.getOntology();
+    // Override default expand options with command-line options
+    ExpandOperation.ExpandConfig config = new ExpandOperation.ExpandConfig();
```

Comment:
For other commands, we've done a `getDefaultOptions()`:
https://github.com/ontodev/robot/blob/2efe4614357eb90c41b45abd92b2802359b1d020/robot-core/src/main/java/org/obolibrary/robot/ExtractOperation.java#L49-L58

Then override them:
https://github.com/ontodev/robot/blob/2efe4614357eb90c41b45abd92b2802359b1d020/robot-command/src/main/java/org/obolibrary/robot/ExtractCommand.java#L163-L168

I don't see a problem with this method, but I'm wondering if - for consistency's sake - we should standardize how we handle default options? We've never really made this a "rule" in the past. @jamesaoverton

Reply:
One of my goals there was to be a bit more type-safe vs. map of strings. But the object could be returned from a method `getDefaultOptions` if desired.